src/game_inventory.cpp

```diff
@@ -134,18 +134,114 @@ int game::inv_for_id( const itype_id &id, const std::string &title )
     }, string_format( _( "You don't have a %s." ), item::nname( id ).c_str() ) );
 }
 
-int game_menus::inv::take_off( player &p )
+class armor_inventory_preset: public inventory_selector_preset
 {
-    return g->inv_for_filter( _( "Take off item" ), [ &p ]( const item & it ) {
-        return p.is_worn( it );
-    }, _( "You don't wear anything." ) );
+    public:
+        armor_inventory_preset( const std::string &color ) : color( color ) {
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->get_encumber() );
+            }, _( "ENCUMBRANCE" ) );
+
+            append_cell( [ color ]( const item_location & loc ) {
+                return loc->get_storage() > 0 ? string_format( "<%s>%s</color>", color,
+                        format_volume( loc->get_storage() ) ) : std::string();
+            }, _( "STORAGE" ) );
+
+            append_cell( [ color ]( const item_location & loc ) {
+                return string_format( "<%s>%d%%</color>", color, loc->get_coverage() );
+            }, _( "COVERAGE" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->get_warmth() );
+            }, _( "WARMTH" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->bash_resist() );
+            }, _( "BASH" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->cut_resist() );
+            }, _( "CUT" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->acid_resist() );
+            }, _( "ACID" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->fire_resist() );
+            }, _( "FIRE" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_number_string( loc->get_env_resist() );
+            }, _( "ENV" ) );
+        }
+
+    private:
+        std::string get_number_string( int number ) const {
+            return number ? string_format( "<%s>%d</color>", color, number ) : std::string();
+        }
+
+        const std::string &color;
```

Comment:
This seems a bit dangerous. Can you *ensure* that the referred string object stays valid through out the lifetime of this object? Changing this into a value (not reference) would make it safer, and would most likely not cost any performance.